### ISSUE TITLE
tracking issue for mapping all Galaxy file formats

### ISSUE THREAD

**Issue by mr-c** [edamontology/edamontology#85]
More to come later.

**Comment by joncison** [comment]
:+1:   I've seen various files / attempts and I'm not sure how to proceed: @hmenager ?

**Comment by joncison** [comment]
Relevant links:

https://docs.google.com/spreadsheets/d/1vptTpJvQfb5FOtRJOajjftUkL9aZ0eeVdhUPvd9h5wY/edit#gid=0

https://gist.github.com/jmchilton/0c5bee8ab199b2f2deb9/revisions

**Comment by hmenager** [comment]
This mapping is now being integrated in the Galaxy source code, in the bio.tools fork:
https://github.com/bio-tools/galaxy
This is based on John Chilton's work

**Comment by mr-c** [comment]
Great! Is there a (standing) pull request we can look at?

În Joi, 3 mar. 2016, 15:35 Hervé Ménager, <notifications@github.com> a
scris:

**Comment by hmenager** [comment]
No PR yet, fork is at github.com/bio-tools/galaxy.

Le ven. 4 mars 2016 07:34, Michael R. Crusoe <notifications@github.com> a
écrit :

**Comment by mr-c** [comment]
Do you mind if I create a standing pull request marked '[WIP]' ?  It would
make it easy to see the changes.

**Comment by joncison** [comment]
I'm OK with this, but @hmenager is best placed to say.  Hervé ?

**Comment by hmenager** [comment]
Sorry for the delayed answer. As far as I'm concerned, I don't mind at all.
Any opinion, @bgruening?

**Comment by bgruening** [comment]
Sounds like a good idea.

**Comment by mr-c** [comment]
Done, see https://github.com/galaxyproject/galaxy/pull/189

**Comment by joncison** [comment]
@hmenager @mr-c @bgruening : next release will include more data formats, is there any more we can do on the EDAM side re supporting use of EDAM in Galaxy?  Are additional (but not yet requested) formats needed?    Just giving this a nudge guys ...

cc @matuskalas

**Comment by bgruening** [comment]
Yes, we have a bunch of formats were no EDAM entry exists (or I was not able to find it :().

**Comment by joncison** [comment]
Thanks! If someone can provide a list of these we can make a start, ideally providing:
* Canonical format name
* Synonym(s)
* Short description
* Link to documentation / formal specification
* Proposed EDAM parent and/or type of data identified

as per _e.g._ https://github.com/edamontology/edamontology/issues/211, that would be super-helpful and make their incorporation in EDAM go so much faster.

**Comment by joncison** [comment]
Hi @mr-c @bgruening @hmenager - just giving this a nudge, if you guys know of Galaxy formats not covered yet by EDAM, please let me know about them and I'll add them in.  We now have an "EDAM Format concept request" template in GitHub which makes things a bit easier ...

**Comment by matuskalas** [comment]
I'm taking this up now, as a CoFest & after project.

I'll go through all the files in https://github.com/galaxyproject/galaxy/tree/dev/lib/galaxy/datatypes and will:
- Uncover the hierarchy of datatype classes in Galaxy, and see how it matches with EDAM.

Then iterate again through the hierarchy and at the same time:
- Add & update the EDAM annotations in those .py files where applicable. **Issue pull requests VERY often!**
- Experimental sideline: Look at the sniffer code and think about whether we could include anything relevant to EDAM (maybe definable outside of Galaxy for new not-yet-in-Galaxy formats?)
- Add & update formats and their attributes in EDAM where applicable. **_N.B._:** _file_ext_ in Galaxy is sometimes more like a short name than a file extension proper.

**Comment by hmenager** [comment]
check out
https://usegalaxy.org/api/datatypes/edam_formats
https://usegalaxy.org/api/datatypes/edam_data

**Comment by hmenager** [comment]
https://github.com/galaxyproject/galaxy/blob/master/scripts/edam_mapping.py
https://github.com/galaxyproject/galaxy/blob/master/lib/galaxy/webapps/galaxy/buildapp.py
https://github.com/galaxyproject/galaxy/blob/dev/config/datatypes_conf.xml.sample

**Comment by matuskalas** [comment]
### Updated plan after more exploration & advice

The holy grail is https://github.com/galaxyproject/galaxy/blob/dev/config/datatypes_conf.xml.sample

- EDAM annotations can also be directly there if there is no specific class (`subclass="true"`)
- `extension=` there means a display name, shown in the Upload file dropdown
- `file_ext` in the .py classes is supposed to be the extension given to a file when downloaded from Galaxy, and supposedly can various classes have the same file extension! (check in .py files)
- datatypes without `display_in_upload=` are probably deprecated (_e.g._ xls, instead see excel.xls and xslx)
- see what to do with ones that `display_in_upload=false`
- _N.B.:_ **Don't include compression formats into EDAM, we'll use a revamped version of WSIO for that, in line with the new Galaxy implementation decisions.**

**Comment by matuskalas** [comment]
Note: The datatype directories in repos with Galaxy tools (_e.g._ https://github.com/galaxyproject/tools-iuc/tree/master/datatypes or https://github.com/bgruening/galaxytools/tree/master/datatypes) are something we should ignore here, as the idea is to include all datatypes into the core Galaxy code.

**Comment by matuskalas** [comment]
Btw.:
Sniffs can be found here https://github.com/galaxyproject/galaxy/search?q=sniff_prefix&unscoped_q=sniff_prefix
The question is, how many of them could be accomplished by regeces and how readable those regeces would be.

**Comment by joncison** [comment]
Galaxy includes many of the tabular formats, so it may make sense to do this simultaneously with https://github.com/edamontology/edamontology/issues/284 (and also ideally with the curation of the Galaxy data types themselves)

**Comment by rioualen** [comment]
Hello, 

Is this issue still relevant? If so, can it be broken down into actionnable items? Otherwise, can be closed.
